### PR TITLE
fix: DownSelfQuarantinedByRemote should not be triggered by harmless quarantine, #31095

### PR DESCRIPTION
**Pull request by patriknw**
* the harmless flag was taken into account in the immediate Quarantined message but it was always sent for later attempts from InboundQuarantineCheck

References #31095

### CODE REVIEW

File: akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala [hunk 1]
```diff
@@ -46,58 +45,58 @@ class DowningWhenOtherHasQuarantinedThisActorSystemMultiJvmNode2
...
+
   "Cluster node downed by other" must {
 
     "join cluster" taggedAs LongRunningTest in {
-      awaitClusterUp(first, second, third)
+      awaitClusterUp(first, second, third, fourth)
       enterBarrier("after-1")
     }
 
-    "down itself" taggedAs LongRunningTest in {
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala:61]: Note that normal downing will result in harmless=true.

File: akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala [hunk 2]
```diff
@@ -113,15 +112,56 @@ abstract class DowningWhenOtherHasQuarantinedThisActorSystemSpec
         cluster.shutdown()
       }
 
+      runOn(first) {
+        expectNoMessage(1.second) // no ThisActorSystemQuarantinedEvent
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/DowningWhenOtherHasQuarantinedThisActorSystemSpec.scala:116]: this wait was the only thing needed to reproduce the issue

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
I have run most relevant multi-jvm tests locally

**Pull request merged**